pkg/skaffold/runner/deployer.go

```diff
@@ -41,7 +41,8 @@ type deployerCtx struct {
 }
 
 func (d *deployerCtx) GetKubeContext() string {
-	if d.deploy.KubeContext != "" {
+	// if the kubeContext is not overridden by CLI flag or env. variable then use the value provided in config.
```

Comment:
nit:  It might make sense to add some E2E/integration test that can verify that the ordering is preserved going forward